att_err_mon.py

```diff
@@ -161,71 +153,105 @@ def att_err_time_plots(ref_data, msd_data, min_dwell_time=1000, outdir='.'):
         plt.setp(ax2.get_yticklabels(), fontsize=7)
         plt.setp(ax1.get_xticklabels(), rotation=0)
         plt.setp(ax1.get_xticklabels(), horizontalalignment='center')
-        plt.savefig(os.path.join(outdir, '{}_err_vs_time.png'.format(ax)))
+        plt.savefig(os.path.join(outdir, f'{ax}_err_vs_time.png'))
 
 
-def att_err_hist(ref_data, msd_data, label=None, min_dwell_time=1000, outdir='.'):
-    ref_data = ref_data[(ref_data['timestop'] - ref_data['time']) >= min_dwell_time]
-    msd_data = msd_data[(msd_data['timestop'] - msd_data['time']) >= min_dwell_time]
+def att_err_hist(ref_data, recent_data, label=None, min_dwell_time=1000, outdir='.'):
+    ref_data = ref_data[ref_data['dwell_duration'] >= min_dwell_time]
+    recent_data = recent_data[recent_data['dwell_duration'] >= min_dwell_time]
+
+    # Grab the date of the start and stop of the two data sets.
+    d0_str = ref_data['date'][0][0:8]
+    d1_str = ref_data['date'][-1][0:8]
+    d2_str = recent_data['date'][0][0:8]
+    d3_str = recent_data['date'][-1][0:8]
+
     for i, ax in enumerate(['roll', 'point'], 1):
-        fig = plt.figure(figsize=(5, 3.5))
+        plt.figure(figsize=(5, 3.5))
         bin_width = .05
-        lim = np.max([1.4, np.max(msd_data['point_err'])])
+        lim = np.max([1.4, np.max(recent_data['point_err'])])
         if ax == 'roll':
             bin_width = .25
-            lim = np.max([12, np.max(msd_data['roll_err'])])
+            lim = np.max([12, np.max(recent_data['roll_err'])])
```

Comment:
Does there need to be a mechanism in place to prevent this auto-scaling from reacting to one bad (large) data point? In the previous version the plots just became useless for this reason.

One idea might be to keep a count of points that are "too big", clip the points within range, and if count_big > 0 then include some text.

Reply:
Indeed, I thought I somewhat had that by the "outlier" mechanism in the calling method.

The problem with this monitor in general was, I thought, that we aren't sure if we are interested in changes in the distribution on a small scale, or increase in frequency of outliers, or some other behavior.  With that in mind, I'm not sure about the best plot limits.  But the calling code borrowed from kalman_watch to make a list of outliers for review.

Reply:
OK, I didn't notice that about the outliers, I just ran the code, saw that it worked and looked at the output page. (Would be good to get more formal with your PR descriptions and call out new things like that at the top level possibly in an itemized list, and then include your testing as a separate section).

Have you tested the outliers code?

I think the way this current code is written you will still autoscale to the size of the worst outlier, thus getting potentially back into the situation of useless plots.  But I also don't know if you have addressed the processing problems that allowed for bad data points to get into the plotted dataset.

Reply:
The outliers code is working nicely to pick up 47190.  I assume it showed up on your output page?  

I tried to pick some kind of reasonable compromise on the limits for the outliers, but the problem with this kind of code is that we still don't know what weird things will be interesting. And yes, it will still autoscale if we have large values that are not otherwise anomalous.  I don't blame the code that much for previously including MUPS test fires (I don't think I can filter those with kadi).  With a rough understanding that *maybe* those (and our recent huge attitude error going to safe mode) are the only kinds of bad data, the new outlier printing might be enough.  

I also considered using the method you have in the aimpoint code by which you put placeholder points for points that have left the plot.  In the end I thought we didn't care enough for that complexity.